gh-104786: Remove kwargs-based TypedDict creation (#104891)
python/cpython: Doc/library/typing.rst

      assert Point2D(x=1, y=2, label='first') == dict(x=1, y=2, label='first')

>    There is an alternative way to create a ``TypedDict``
>    using a literal :class:`dict` as the second argument::

Reviewer: ```suggestion
   An alternative way to create a ``TypedDict`` is by using
   function-call syntax. The second argument must be a literal :class:`dict`::
```

      Point2D = TypedDict('Point2D', {'x': int, 'y': int, 'label': str})

   This functional syntax allows defining keys which are not valid
   :ref:`identifiers <identifiers>`, for example because they are

>    keywords or contain hyphens.
>    Example::

Reviewer: ```suggestion
   keywords or contain hyphens::
```

      assert Point2D(x=1, y=2, label='first') == dict(x=1, y=2, label='first')

>    To allow using this feature with older versions of Python that do not

Reviewer 2: "versions that do not support PEP 526" means 3.5 or lower, which we no longer care about. We should rewrite this text to motivate the feature only with case for keys that are not valid identifiers (below) and omit mention of PEP 526.

Author: > using a literal :class:`dict` as the second argument

I'm wondering if I should also mention that you can pass an iterable as the second argument and not just a dictionary (same as the `dict` constructor)

Reviewer 2: Like this?

```
>>> TD = TypedDict("TD", iter([("a", int)]))
>>> TD.__annotations__
{'a': <class 'int'>}
```

We shouldn't mention that as type checkers won't support it.

Reviewer: Yup. The runtime is very duck-type-y, but PEP-589 specifically says:

> A type checker is only expected to accept a dictionary display expression as the second argument to TypedDict

https://peps.python.org/pep-0589/#alternative-syntax

So we shouldn't advertise that you can use other iterables for the second second argument :)

>       Point2D = TypedDict('Point2D', x=int, y=int, label=str)

Reviewer: We should also delete this bullet point, since it's no longer valid

   .. versionchanged:: 3.13

>       Removed support for the keyword-argument method of creating ``TypedDict``.

Reviewer: Could you move this note down below the other `.. versionchanged` notes? I.e., just below this line here:

https://github.com/python/cpython/blob/fe77a99fc8b549a8bf9ccbc5485fe5ea9bcf47b9/Doc/library/typing.rst?plain=1#L1943-L1944
